### PR TITLE
cEP-0012: coala's Command Line Interface

### PR DESCRIPTION
**Pull request by Nosferatul**
This cEP describe the design of coala's Command Line Interface(the action
selection screen). It is intended to refactor the design.

Closes https://github.com/coala/cEPs/issues/76

### CODE REVIEW

**Comment by Udayan12167** [comment]
Hey can I request mockups here for the interface? Maybe I'm reviewing too early if you plan to add please ignore.

**Comment by Nosferatul** [comment]
Sure, I think I'll add them tomorrow. I'm working on it

File: cEP-0010.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+|Title   |coala's Command Line Interface                    |
+|Authors |Alexandru-Sorin Maxim <alexandrus.maxim@gmail.com>|
+|Status  |Proposed                                          |
+|Type    |Feature                                           |
+
+
+Abstract
+--------
+
+This cEP describes the design of coala's Command Line Interface (the action selection screen). It intended to refactor the designed released as part of coala 0.8.
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: intends

Comment (Adrianzatreanu) [inline cEP-0010.md]: i think its "the design released"

File: cEP-0010.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+Adds functionality alongside with a better interaction with the user. The final goal is coala CLI to be more friendly, easier to read and use, more functionality depending on what user wants!
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: is for*

Comment (Adrianzatreanu) [inline cEP-0010.md]: last part after the last comma doesnt even have a verb

File: cEP-0010.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+1. Remove most of the text from the action selection screen
+2. Use letters instead of numbers for selecting actions
+3. Allow chaining actions in a single input
+4. Add a 'selection action for all results' option
+5. Make ApplyPatchAction log the applied patch
+6. Reprompt after running actions only if the action says so
+7. Set 'Do nothing' as the default action
+
+The below section describes why we should make these changes.
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: the section below* sounds better

**Comment by Nosferatul** [comment]
@Udayan12167 please check https://docs.google.com/forms/d/e/1FAIpQLSd-E9cmq6FTJy01kLiiO0WrEcYAgIIpLjwHxTiYzZVjPo5xag/viewform?usp=sf_link

**Comment by Nosferatul** [comment]
@Adrianzatreanu @yukiisbored please review

File: cEP-0010.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+[    ] Choose an action:
+
+If an user will press 'C':
+
+[    ] Enter the letters of the patches you want to apply('-' between any letter): a-i-n
+
+Any combination of letters (upper/lower) will be accepted (e.g: a-I-N).
+If one of the action requested isn't possible, an warning will be printed (after all possible actions are applied) with the letter that couldn't be satisfied.
+
+### Add a 'select action for all results' option
```
Comment (yukiisbored) [inline cEP-0010.md]: Does all action refers to all of the action inside the section/bear or all of them regardless of which section/bear?

Comment (Nosferatul) [reply cEP-0010.md]: I was thinking about inside the section/bear

File: cEP-0010.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+|Title   |coala's Command Line Interface                    |
+|Authors |Alexandru-Sorin Maxim <alexandrus.maxim@gmail.com>|
+|Status  |Proposed                                          |
+|Type    |Feature                                           |
+
+
+Abstract
+--------
+
+This cEP describes the design of coala's Command Line Interface (the action selection screen). It intended to refactor the design released as part of coala 0.8.
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: intends* ?

File: cEP-0010.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+Adds functionality alongside with a better interaction with the user. The final goal is for coala CLI to be more friendly, easier to read and use. Adds more functionality depending on what user wants!
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: This cEP adds..

Comment (Adrianzatreanu) [inline cEP-0010.md]: cut the last phrase, sounds cheesy

File: cEP-0010.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+Changes
+-------
+
+The following changes shall be made to the CLI:
+
+1. Remove most of the text from the action selection screen
+2. Use letters instead of numbers for selecting actions
+3. Allow chaining actions in a single input
+4. Add a 'selection action for all results' option
+5. Make ApplyPatchAction log the applied patch
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: enclose ApplyPatchAction in double backticks

File: cEP-0010.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+7. Set 'Do nothing' as the default action
+
+The section below describes why we should make these changes.
+
+Why
+---
+
+### Remove most of the text from the action selection screen
+
+Most of the people who use a CLI program do not read the whole output, only a small part of it. Also, a large output from a command could tire them and could make them stop using our product We want to remove the redundant information, keeping only the necessary information.
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: such long lines. please restrict the length of a line to 80 chars.

Comment (Adrianzatreanu) [inline cEP-0010.md]: period after "product"

Comment (Adrianzatreanu) [inline cEP-0010.md]: "our product" sounds bad, make it more like "stop using the application" or so?

File: cEP-0010.md [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+
+### Use letters instead of numbers for selecting actions
+
+Easier to remember an action based on the letters instead of numbers. (Do (n)othing).
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: an action based on a letter*

File: cEP-0010.md [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+
+### Allow chaining actions in a single input
+
+Users could apply more than one patch in a single input. The idea here is to offer more flexibility to our users. Instead of just 1 number, the user could enter more actions.
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: more actions, chained by commas etc etc

File: cEP-0010.md [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+
+The section below describe how we should make these changes.
+
+How
+---
+
+### Remove most of the text from the action selection screen
+
+Most of the text that should be changed is in the 'ConsoleInteraction.py' file. We should change that based on the feedback requested with
+Google Form.
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: feedbacked gathered on the Google Form.

**Comment by Adrianzatreanu** [comment]
i reviewed a bit of it but there's something i dont like at all. youre using a lot of "we will, we should, I will". This is not how a document, e.g. enhancement proposal should look like (nobody cares who wrote it).
Take a look at cep 02 for example, check how everything is written (third person):
This proposal aims to; A new class will be created; The bears will be handled by; etc, you dont see "We will create a new class". feel the difference?

File: cEP-0010.md (hunk 5 above)
Comment (yukiisbored) [reply cEP-0010.md]: I think "It is intended ..." fits more.

Comment (yukiisbored) [inline cEP-0010.md]: Also, Please limit to 80 chars.

Comment (yukiisbored) [inline cEP-0010.md]: Please limit to 80 chars

Comment (yukiisbored) [inline cEP-0010.md]: Please limit to 80 characters

File: cEP-0010.md (hunk 6 above)
Comment (yukiisbored) [inline cEP-0010.md]: Please limit to 80 characters

File: cEP-0010.md [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+
+Easier to remember an action based on the letters instead of numbers. (Do (n)othing).
+
+### Allow chaining actions in a single input
+
+Users could apply more than one patch in a single input. The idea here is to offer more flexibility to our users. Instead of just 1 number, the user could enter more actions.
+
+### Add a 'select action for all results' option
+
+Now users have the possibility to choose at every result an action. We want to add the possibility to choose an action for all results.
```
Comment (yukiisbored) [inline cEP-0010.md]: I find this to be quite confusing. I think "to choose an action for every results" would be a better choice.

File: cEP-0010.md [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+
+### Reprompt after running actions
+
+We should have an option that lets user choose whether they want another action after the current action is done. This way, users will have the flexibility in their hands.
```
Comment (yukiisbored) [inline cEP-0010.md]: I don't get what this means nor the use case for this.

File: cEP-0010.md [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+
+Most of the text that should be changed is in the 'ConsoleInteraction.py' file. We should change that based on the feedback requested with
+Google Form.
+
+### Use letters instead of numbers for selecting actions
+
+Usage of letters would make usage of coala faster, as one could just remember
+that `a` stands for 'apply patch' for instance, and they would not have to check
+the list for every result to see what number 'apply patch' has been assigned.
+Masive changes should take place in 'result_actions' directory where are kept all the actions.
```
Comment (yukiisbored) [inline cEP-0010.md]: s/Masive/Massive/

File: cEP-0010.md [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+[    ] Do (N)othing
+[    ] (O)pen file
+[    ] (A)pply the patch
+[    ] Add (i)gnore comment
+[    ] (C)hain actions
+[    ] Choose an action:
+
+If an user will press 'C':
+
+[    ] Enter the letters of the patches you want to apply('-' between any letter): a-i-n
```
Comment (yukiisbored) [inline cEP-0010.md]: Add a space before the first parentheses

File: cEP-0010.md [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+If the user will press 'S':
+
+==== Apply an action for all results ====
+[    ] Do (N)othing
+[    ] (O)pen file
+[    ] (A)pply the patch
+[    ] Add (I)gnore comment
+[    ] Choose an action:
+
+So, let's say that a user wants to add an ignore comment in all files.
```
Comment (yukiisbored) [inline cEP-0010.md]: s/let's/lets/

File: cEP-0010.md [hunk 17]
```diff
@@ -0,0 +1,142 @@
...
+The user should choose '(S)elect action for all results' and after, 'i' option.
```
Comment (yukiisbored) [inline cEP-0010.md]: I think "and choose the 'i' option" would fit better. The comma placement is confusing.

File: cEP-0010.md [hunk 18]
```diff
@@ -0,0 +1,142 @@
...
+This will add an ignore comment in all of the files used in coafile/command without asking another action.
+
+### Reprompt after running actions
+
+This should work if the user choose to (C)hain actions.
+
+[    ] Enter the letters of the patches you want to apply('-' between any letter): a-i-n
```
Comment (yukiisbored) [inline cEP-0010.md]: Please add space before the first parentheses

File: cEP-0010.md [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+[    ] Reprompt after running actions(Y/N)?
```
Comment (yukiisbored) [inline cEP-0010.md]: Same goes for here, add a space before the first parentheses

File: cEP-0010.md [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
+
+[    ] Do (N)othing
+[    ] (O)pen file
+[    ] (A)pply the patch
+[    ] Add (i)gnore comment
+[    ] (C)hain actions
+[    ] (S)elect action for all results
+[    ] Choose an action:
+
+### Set 'Do nothing' as the default action
```
Comment (yukiisbored) [inline cEP-0010.md]: Remove this, Duplication of https://github.com/coala/cEPs/pull/79/files#diff-dc65288ae3b8f48bf19e685ea3460ad6R59

File: cEP-0010.md (hunk 11 above)
Comment (yukiisbored) [reply cEP-0010.md]: Did you mean, "feedback gathered from the Google Form"?

File: cEP-0010.md (hunk 5 above)
Comment (Adrianzatreanu) [reply cEP-0010.md]: or that, still better than "it indended"

File: cEP-0010.md [hunk 21]
```diff
@@ -0,0 +1,152 @@
...
+
+1. Remove most of the text from the action selection screen
+2. Use letters instead of numbers for selecting actions
+3. Allow chaining actions in a single input
+4. Add a 'selection action for all results' option
+5. Make `ApplyPatchAction` log the applied patch
+6. Reprompt after running actions only if the action says so
+7. Set 'Do nothing' as the default action
+
+The section below describes why we should make these changes.
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: why these changes should be done*

File: cEP-0010.md [hunk 22]
```diff
@@ -0,0 +1,152 @@
...
+
+Why
+---
+
+### Remove most of the text from the action selection screen
+
+Most of the people who use a CLI program do not read the whole output, only a
+small part of it. Also, a large output from a command could tire them and could
+make them stop using our application. We want to remove the redundant information,
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: no "we want"

File: cEP-0010.md [hunk 23]
```diff
@@ -0,0 +1,152 @@
...
+keeping only the necessary information.
+
+### Use letters instead of numbers for selecting actions
+
+Easier to remember an action based on a letter instead of numbers. (Do (n)othing).
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: Letters are easier to be remembered rather than numbers.

File: cEP-0010.md [hunk 24]
```diff
@@ -0,0 +1,152 @@
...
+### Allow chaining actions in a single input
+
+Users could apply more than one patch in a single input. The idea here is to
+offer more flexibility to our users. Instead of just 1 number, the user could enter
+more actions.
+
+### Add a 'select action for all results' option
+
+Now users have the possibility to choose at every result an action.
+We want to add the possibility to choose an action for all results.
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: no "we want"

File: cEP-0010.md [hunk 25]
```diff
@@ -0,0 +1,152 @@
...
+
+### Reprompt after running actions
+
+We should have an option that lets user choose whether they want another action after the current action is done. This way, users will have the flexibility in their hands.
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: no "we should"

File: cEP-0010.md [hunk 26]
```diff
@@ -0,0 +1,152 @@
...
+
+### Set 'Do nothing' as the default action
+
+If a user chooses a non-existent action (or the input is wrong) just exit from the program with 'Do nothing'. This is for reuse of 'Do nothing'.
+
+The section below describe how we should make these changes.
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: describes*

File: cEP-0010.md [hunk 27]
```diff
@@ -0,0 +1,152 @@
...
+
+How
+---
+
+### Remove most of the text from the action selection screen
+
+Most of the text that should be changed is in the 'ConsoleInteraction.py' file. We should change that based on the feedback requested with
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: no "we should"

File: cEP-0010.md (hunk 11 above)
Comment (Adrianzatreanu) [reply cEP-0010.md]: ^

File: cEP-0010.md [hunk 28]
```diff
@@ -0,0 +1,152 @@
...
+[    ] Add (i)gnore comment
+[    ] (C)hain actions
+[    ] Choose an action:
+
+If an user will press 'C':
+
+[    ] Enter the letters of the patches you want to apply ('-' between any letter): a-i-n
+
+Any combination of letters (upper/lower) will be accepted (e.g: a-I-N).
+If one of the action requested isn't possible, an warning will be printed (after all possible actions are applied) with the letter that couldn't be satisfied.
```
Comment (Adrianzatreanu) [inline cEP-0010.md]: a warning*

**Review by yukiisbored — CHANGES_REQUESTED** [review]
Please change the cEP number, It's conflicting with @Asnelchristian's cEP.

**Comment by Nosferatul** [comment]
done @Adrianzatreanu @yukiisbored

File: cEP-0012.md [hunk 29]
```diff
@@ -0,0 +1,152 @@
...
+If a user chooses a non-existent action (or the input is wrong) just exit from the program with 'Do nothing'. This is for reuse of 'Do nothing'.
+
+The section below describes how we should make these changes.
+
+How
+---
+
+### Remove most of the text from the action selection screen
+
+Most of the text that should be changed is in the 'ConsoleInteraction.py' file. Do changes based on the feedback requested with
```
Comment (yukiisbored) [inline cEP-0012.md]: I think it should be gathered not requested.

File: cEP-0012.md [hunk 30]
```diff
@@ -0,0 +1,152 @@
...
+7. Set 'Do nothing' as the default action
+
+The section below describes why these changes should be done.
+
+Why
+---
+
+### Remove most of the text from the action selection screen
+
+Most of the people who use a CLI program do not read the whole output, only a
```
Comment (yukiisbored) [inline cEP-0012.md]: I think it should be "does not" since "Most of the people" is third-person

File: cEP-0012.md [hunk 31]
```diff
@@ -0,0 +1,152 @@
...
+This is an option that lets user choose whether they want another action after the current action is done. This way, users will have the flexibility in their hands.
+
+### Set 'Do nothing' as the default action
+
+If a user chooses a non-existent action (or the input is wrong) just exit from the program with 'Do nothing'. This is for reuse of 'Do nothing'.
+
+The section below describes how we should make these changes.
+
+How
+---
```
Comment (yukiisbored) [inline cEP-0012.md:72]: Nothing to fill here or WIP until the next period?

File: cEP-0012.md (hunk 29 above)
Comment (yukiisbored) [reply cEP-0012.md]: Not to mention line 79  uses "gathered"

File: cEP-0012.md (hunk 31 above)
Comment (Nosferatul) [reply cEP-0012.md:72]: I explained below or do you want something else?

Comment (yukiisbored) [reply cEP-0012.md:72]: Well, line 69 said "The section below describes how we should make these changes". So yeah, explain it under this section.

Comment (Nosferatul) [reply cEP-0012.md:72]: Hmm, do you want code? The idea is explained below for every change

File: cEP-0012.md [hunk 32]
```diff
@@ -0,0 +1,152 @@
...
+|Title   |coala's Command Line Interface                    |
+|Authors |Alexandru-Sorin Maxim <alexandrus.maxim@gmail.com>|
+|Status  |Proposed                                          |
+|Type    |Feature                                           |
+
+
+Abstract
+--------
+
+This cEP describes the design of coala's Command Line Interface (the action selection screen). It intended to refactor the design released as part of coala 0.8.
```
Comment (Adrianzatreanu) [inline cEP-0012.md]: it indends*!!!!!

File: cEP-0012.md [hunk 33]
```diff
@@ -0,0 +1,152 @@
...
+7. Set 'Do nothing' as the default action
+
+The section below describes why these changes should be done.
+
+Why
+---
+
+### Remove most of the text from the action selection screen
+
+Most of the people who use a CLI program does not read the whole output, only a
```
Comment (Adrianzatreanu) [inline cEP-0012.md]: do not*

File: cEP-0012.md [hunk 34]
```diff
@@ -0,0 +1,152 @@
...
+
+How
+---
+
+### Remove most of the text from the action selection screen
+
+Most of the text that should be changed is in the 'ConsoleInteraction.py' file. Do changes based on the feedback gathered with
+Google Form.
+
+1. Make changes in the 'ConsoleInteraction.py' based on the feedback gathered with Google Form.
```
Comment (Adrianzatreanu) [inline cEP-0012.md]: what is this 1?

File: cEP-0012.md [hunk 35]
```diff
@@ -0,0 +1,152 @@
...
+Mockup:
+
+[    ] Do (N)othing
+[    ] (O)pen file
+[    ] (A)pply the patch
+[    ] Add (i)gnore comment
+[    ] (C)hain actions
+[    ] Choose an action:
+
+If an user will press 'C':
```
Comment (Adrianzatreanu) [inline cEP-0012.md]: it is "a user"
https://english.stackexchange.com/questions/105116/is-it-a-user-or-an-user

File: cEP-0012.md [hunk 36]
```diff
@@ -0,0 +1,152 @@
...
+
+[    ] Enter the letters of the patches you want to apply ('-' between any letter): a-i-n
+
+Any combination of letters (upper/lower) will be accepted (e.g: a-I-N).
+If one of the action requested isn't possible, a warning will be printed (after all possible actions are applied) with the letter that couldn't be satisfied.
```
Comment (Adrianzatreanu) [inline cEP-0012.md]: actions*

File: cEP-0012.md [hunk 37]
```diff
@@ -0,0 +1,152 @@
...
+
+1. Create a new file 'ChainAction.py' and a new class 'ChainAction' in 'result_actions' directory
+2. Implement the logic in the new class
+3. Parse corecty the input
```
Comment (Adrianzatreanu) [inline cEP-0012.md]: parse the input corRectly*

File: cEP-0012.md [hunk 38]
```diff
@@ -0,0 +1,152 @@
...
+[    ] Add (I)gnore comment
+[    ] Choose an action:
+
+So, let's say that a user wants to add an ignore comment in all files.
+The user should choose '(S)elect action for all results' and after choose the 'i' option.
+This will add an ignore comment in all of the files used in coafile/command without asking another action.
+
+### Reprompt after running actions
+
+This should work if the user choose to (C)hain actions.
```
Comment (Adrianzatreanu) [inline cEP-0012.md]: chooses

File: cEP-0012.md [hunk 39]
```diff
@@ -0,0 +1,152 @@
...
+
+[    ] Enter the letters of the patches you want to apply ('-' between any letter): a-i-n
+[    ] Reprompt after running actions (Y/N)?
+
+If the user choose 'Y', after all the actions are applied, the interface will be reprompted with the possible actions.
```
Comment (Adrianzatreanu) [inline cEP-0012.md]: chooses

File: cEP-0012.md (hunk 31 above)
Comment (yukiisbored) [reply cEP-0012.md:72]: Oh, didn't see that. Sorry :sweat_smile:

**Review by yukiisbored — DISMISSED** [review]

File: cEP-0012.md (hunk 30 above)
Comment (yukiisbored) [inline cEP-0012.md]: nvm, it's correct.

File: cEP-0012.md [hunk 40]
```diff
@@ -0,0 +1,152 @@
...
+that `a` stands for 'apply patch' for instance, and they would not have to check
+the list for every result to see what number 'apply patch' has been assigned.
+
+Do masive changes in 'result_actions' directory where are kept all the actions.
+
+### Allow chaining actions in a single input
+
+Mockup:
+
+[    ] Do (N)othing
```
Comment (yukiisbored) [inline cEP-0012.md]: Please use the multi-line code blocks for the mock-ups. It looks weird in the "Rich text view"

File: cEP-0012.md [hunk 41]
```diff
@@ -0,0 +1,152 @@
...
+1. Create a new file 'ChainAction.py' and a new class 'ChainAction' in 'result_actions' directory
+2. Implement the logic in the new class
+3. Parse corectly the input
+3. Use the new class in 'ConsoleInteraction'
+
+### Add a 'select action for all results' option
+
+Mockup:
+
+[    ] Do (N)othing
```
Comment (yukiisbored) [inline cEP-0012.md]: Please use the multi-line code blocks for the mock-ups. It looks weird in the "Rich text view"

File: cEP-0012.md [hunk 42]
```diff
@@ -0,0 +1,152 @@
...
+[    ] (O)pen file(s)
+[    ] (A)pply the patch
+[    ] Add (i)gnore comment
+[    ] (C)hain actions
+[    ] (S)elect action for all results
+[    ] Choose an action:
+
+If the user will press 'S':
+
+==== Apply an action for all results ====
```
Comment (yukiisbored) [inline cEP-0012.md]: Please use the multi-line code blocks for the mock-ups. It looks weird in the "Rich text view"

File: cEP-0012.md [hunk 43]
```diff
@@ -0,0 +1,152 @@
...
+
+So, let's say that a user wants to add an ignore comment in all files.
+The user should choose '(S)elect action for all results' and after choose the 'i' option.
+This will add an ignore comment in all of the files used in coafile/command without asking another action.
+
+### Reprompt after running actions
+
+This should work if the user chooses to (C)hain actions.
+
+[    ] Enter the letters of the patches you want to apply ('-' between any letter): a-i-n
```
Comment (yukiisbored) [inline cEP-0012.md]: Please use the multi-line code blocks for the mock-ups. It looks weird in the "Rich text view"

File: cEP-0012.md [hunk 44]
```diff
@@ -0,0 +1,152 @@
...
+[    ] Reprompt after running actions (Y/N)?
+
+If the user chooses 'Y', after all the actions are applied, the interface will be reprompted with the possible actions.
+
+[    ] Do (N)othing
```
Comment (yukiisbored) [inline cEP-0012.md]: Please use the multi-line code blocks for the mock-ups. It looks weird in the "Rich text view"

File: cEP-0012.md [hunk 45]
```diff
@@ -0,0 +1,152 @@
...
+[    ] (O)pen file
+[    ] (A)pply the patch
+[    ] Add (i)gnore comment
+[    ] (C)hain actions
+[    ] Choose an action:
+
+If a user will press 'C':
+
+[    ] Enter the letters of the patches you want to apply ('-' between any letter): a-i-n
```
Comment (yukiisbored) [inline cEP-0012.md]: Please use the multi-line code blocks for the mock-ups. It looks weird in the "Rich text view"

**Comment by yukiisbored** [comment]
ack https://github.com/coala/cEPs/commit/ed65db9ff3441eb3eeeb39035eed10875aa27435

**Comment by yukiisbored** [comment]
@rultor Please merge xoxo

**Pull request merged**